src/targets/mobile/containers/RatingModal.jsx

```diff
@@ -0,0 +1,236 @@
+/* global cozy AppRate */
+import React, { Component } from 'react'
+import Modal, { ModalContent } from 'cozy-ui/react/Modal'
+import { connect } from 'react-redux'
+import { softwareID, softwareName } from '../lib/cozy-helper'
+import withPersistentState from '../lib/withPersistentState'
+
+import styles from '../styles/rating'
+
+const SCREEN_ENJOY = 'SCREEN_ENJOY'
+const SCREEN_FEEDBACK = 'SCREEN_FEEDBACK'
+const ALERT_RATING = 'ALERT_RATING'
+
+const BUTTON_INDEX_RATE = 1
+const BUTTON_INDEX_LATER = 2
+
+const FEEDBACK_EMAIL = 'support@cozycloud.cc'
+const PROMPT_AFTER_BOOTS = 5
+
+// RatingModal is the base component
+class RatingModal extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      screen: SCREEN_ENJOY
+    }
+  }
+
+  async onUserReply (enjoyed) {
+    if (enjoyed) {
+      try {
+        const { t } = this.context
+        const buttonIndex = await promptRating({
+          title: t('mobile.rating.rate.title'),
+          message: '',
+          yes: t('mobile.rating.rate.yes'),
+          no: t('mobile.rating.rate.no'),
+          later: t('mobile.rating.rate.later'),
+          softwareID,
+          softwareName
+        })
+
+        if (buttonIndex === BUTTON_INDEX_LATER) {
+          this.props.alert('mobile.rating.alert.later')
+          this.props.showLater()
+        }
+        else if (buttonIndex === BUTTON_INDEX_RATE) {
+          this.props.alert('mobile.rating.alert.rated')
+          this.props.dontShowAgain()
+        }
+        else {
+          this.props.alert('mobile.rating.alert.declined')
+          this.props.dontShowAgain()
+        }
+      } catch (e) {
+        this.props.dontShowAgain()
+        throw e
+      }
+    } else {
+      this.setState({ screen: SCREEN_FEEDBACK })
+    }
+  }
+
+  onCloseFeedback (gaveFeedback) {
+    if (gaveFeedback) this.props.alert('mobile.rating.alert.feedback')
+    this.props.dontShowAgain()
+  }
+
+  render () {
+    const { screen } = this.state
+
+    return screen === SCREEN_ENJOY ? <EnjoyCozy onReply={this.onUserReply.bind(this)} /> : <FeeedbackForm onClose={this.onCloseFeedback.bind(this)} />
+  }
+}
+
+// sub-components
+class EnjoyCozy extends Component {
+  render ({ onReply }) {
```

Comment:
Do we want to stay React compatible? If yes, we cannot use parameter in `render()`.